### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Oseun12**
Updated set up to use Enter-key to add, Cancel btn to stop the process during editing and also Modal to show confirmation before deleting.

### CODE REVIEW

File: pages/components/Todolist.tsx [hunk 1]
```diff
@@ -21,7 +23,17 @@ const Todolist = () => {
...
+
+    const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'Enter') {
+            if (editIndex > -1) {
+                saveTodo();
+            } else {
+                addTodo();
+            }
+        }
+    };
```
Comment (unclebay143) [inline pages/components/Todolist.tsx:36]: I mean to remove this function, we can achieve it through combination of form and button

File: pages/components/Todolist.tsx [hunk 2]
```diff
@@ -55,31 +81,39 @@ const Todolist = () => {
...
-                    <button onClick={saveTodo} className="btn ml-2">
-                        Save
-                    </button>
-                ) : (
-                    <button onClick={addTodo} className="btn ml-2">
-                        Add
-                    </button>
-                )}
-            </div>
+            <form onSubmit={(e) => e.preventDefault()}>
```
Comment (unclebay143) [inline pages/components/Todolist.tsx:84]: let's pass the `addTodo` or `editTodo` here dynamically

File: pages/components/Todolist.tsx [hunk 3]
```diff
@@ -55,31 +81,39 @@ const Todolist = () => {
...
+                        onChange={(e) => setNewTodo(e.target.value)}
+                        onKeyPress={handleKeyPress}
+                        placeholder="Enter a new task..."
+                        className="input input-bordered w-full max-w-xs text-black bg-white"
+                    />
+                    {editIndex > -1 ? (
+                        <>
+                            <button onClick={saveTodo} className="btn ml-2">
+                                Save
+                            </button>
```
Comment (unclebay143) [inline pages/components/Todolist.tsx:98]: should have `type` or `submit`

File: pages/components/Todolist.tsx [hunk 4]
```diff
@@ -55,31 +81,39 @@ const Todolist = () => {
...
+                            <button onClick={cancelEdit} className="btn ml-2 btn-warning">
+                                <FaTimes />
+                            </button>
+                        </>
+                    ) : (
+                        <button onClick={addTodo} className="btn ml-2">
```
Comment (unclebay143) [inline pages/components/Todolist.tsx:104]: should have `type` or `submit`

File: pages/components/Todolist.tsx (hunk 1 above)
Comment (Oseun12) [reply pages/components/Todolist.tsx:36]: Done now sir

**Review by unclebay143 — APPROVED** [review]

**Pull request merged**